felix/docker-image/Dockerfile

```diff
@@ -67,7 +67,7 @@ RUN update-alternatives --set ip6tables /usr/sbin/ip6tables-legacy
 
 # Felix is built with RHEL/UBI and links against libpcap.so.1. We need this symbolic link
 # until Debian changes the soname from .0.8 to .1.
-RUN ln -s $(readlink /usr/lib/*-linux-gnu/libpcap.so.0.8) /usr/lib/x86_64-linux-gnu/libpcap.so.1
+RUN ln -s $(readlink /usr/lib/*-linux-gnu/libpcap.so.0.8) lib$(ldconfig -v 2>/dev/null | grep -Po '/lib/(.*): ' | sed -e 's/: //')/libpcap.so.1
```

Comment:
Just realized that my point (4) is bogus - I was thinking of awk.

Reply:
@nelljerram For context, this change is reverting to how this used to work (with only a small adjustment to account for the image being debian 12 instead of debian 11)

Here's the old incantation: https://github.com/projectcalico/calico/blob/release-v3.28/felix/docker-image/Dockerfile#L69

Here's the comment where @hjiawei requested I revert to this: https://github.com/tigera/calico-private/pull/7767#discussion_r1674833185

That said..

> Can it be right to create the symlink in the working directory, instead of in /usr/lib/x86_64-linux-gnu?

Correct - this was a typo on my part, good catch!

> Why the brackets around .* ?

Yeah, probably don't need those.

> What is the ldconfig intended to do? I think it's worth commenting this.

I believe it's just finding the correct path based on the arch. Not 100% sure, I didn't write this originally and I'm just tuning it up for the new base image.

> Doesn't entirely work for me here:

The output of ldconfig varies based on distro and version - the only reason I'm changing this is because the old debian 11 command no longer worked when I switched to deb 12 (needed for nftables work). I'm not too surprised this command doesn't work in your env, but I'd expect it to work if you ran inside of a debian 12 container.

That said, I agree the command you suggested looks much nicer, so I'll swap to that. Thank you!